lib/src/klio/transforms/decorators.py

```diff
@@ -183,10 +183,28 @@ def __serialize_klio_message_generator(
 
     else:
         if isinstance(payload, types.GeneratorType):
-            for pl in payload:
-                yield from __from_klio_message_generator(
-                    self, kmsg, pl, incoming_item
+            try:
+                for pl in payload:
+                    yield from __from_klio_message_generator(
+                        self, kmsg, pl, incoming_item
+                    )
+            # This exception will the the pl item if it was an Exception
+            except Exception as err:
+                func_path = self.__class__.__name__ + "." + meth.__name__
+                log_msg, exc_info = __get_user_error_message(
+                    err, func_path, kmsg
                 )
+                self._klio.logger.error(log_msg, exc_info=exc_info)
+                # This will catch an exception present in the generator
+                # containing items yielded in a job's process function.
```

Comment:
Technically, a function/method that's decorated with `@handle_klio` - not necessarily a DoFn's process method. Just to be clear in the comments / not to confused future us as maintainers 😆